cpp/src/mtmg/vertex_result.cu

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cugraph/detail/utility_wrappers.hpp>
+#include <cugraph/graph_functions.hpp>
+#include <cugraph/mtmg/vertex_result.hpp>
+#include <cugraph/vertex_partition_device_view.cuh>
+
+#include <detail/graph_partition_utils.cuh>
+
+#include <thrust/gather.h>
+
+namespace cugraph {
+namespace mtmg {
+
+template <typename result_t>
+template <typename vertex_t, typename edge_t, bool store_transposed, bool multi_gpu>
+rmm::device_uvector<result_t> vertex_result_t<result_t>::gather(
+  handle_t const& handle,
+  raft::device_span<vertex_t const> vertices,
+  cugraph::mtmg::graph_view_t<vertex_t, edge_t, store_transposed, multi_gpu> const& graph_view,
+  std::optional<cugraph::mtmg::renumber_map_t<vertex_t>>& renumber_map)
```

Comment:
Can't this be const?

Reply:
And we may consider defining a MTMG span or view object instead of passing a constant reference of owning objects to be consistent with the rest of cugraph.

Reply:
Added view objects in the latest push.  The view objects are const.